src/lib/fileutils.ts

```diff
@@ -470,6 +471,29 @@ export const updateTriggerBranchesForServiceBuildAndUpdatePipeline = (
   );
 };
 
+/**
+ * Appends a variable group an Azure pipeline yaml
+ * @param dir The directory where the pipeline yaml file is
+ * @param pipelineFile The name of the pipeline yaml file
+ * @param variableGroupName The name of the variable group to be added
+ */
+export const appendVariableGroupToPipelineYaml = (
```

Comment:
It looks like the PR is still work-in-progress. If so, instead of updating the pipeline yaml after creation, how about generating the file with variable group name when created initially? For ex: https://github.com/CatalystCode/spk/blob/0e0ba83cf04107769bf328329ec3576f236343d2/src/lib/fileutils.ts#L151

Reply:
This is out of scope for this task.